### PR TITLE
components/engine_test: fix double-lock in open_tablet

### PR DESCRIPTION
**Pull request by BurtonQin**
### What is changed and how it works?

Issue Number: Ref #13186

What's Changed:

```
Restrict `registry.lock()` in a scope so that it cannot double-lock with another `registry.lock()` in `open_tablet_raw`->`create_table`
Remove the redundant `reg.insert`, which has been called in `create_tablet`.
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

Side effects

### Release note 

```release-note
Fix a double-lock bug in components/engine_test
```

### CODE REVIEW

File: components/engine_test/src/lib.rs [hunk 1]
```diff
@@ -261,9 +261,11 @@ pub mod kv {
         }
 
         fn open_tablet(&self, id: u64, suffix: u64) -> Result<KvTestEngine> {
-            let reg = self.registry.lock().unwrap();
-            if let Some(db) = reg.get(&(id, suffix)) {
-                return Ok(db.clone());
+            {
+                let reg = self.registry.lock().unwrap();
+                if let Some(db) = reg.get(&(id, suffix)) {
```
Comment (Connor1996) [inline components/engine_test/src/lib.rs:266]:
```suggestion
                if let Some(db) = self.registry.lock().unwrap().get(&(id, suffix)) {
```

**Review by Connor1996 — APPROVED** [review]
Thanks for your finding. LGTM

**Review by tonyxuqqi — APPROVED** [review]

**Pull request merged**